### PR TITLE
Fixed tests

### LINKED ISSUE #45: Broken tests
Running the tests on both Py27 and 37 yielded 4 errors on the mocking of `library.degoss.open`.

I know they are passing in the official build but they failed on both 2.7 and 3.7 for me. Changing the patching to use `mock_open` made it pass on both versions.

```
.............E..EEE....
======================================================================
ERROR: test_install (__main__.DegossTestCase)
Tests that degoss can install Goss successfully.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1193, in patched
    arg = patching.__enter__()
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1268, in __enter__
    original, local = self.get_original()
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1242, in get_original
    "%s does not have the attribute %r" % (target, name)
AttributeError: <module 'library.degoss' from '/Users/nicholas/src/ansible-role-degoss/library/degoss.py'> does not have the attribute 'open'

======================================================================
ERROR: test_run_tests_error (__main__.DegossTestCase)
Tests that degoss can handle error cases when running tests.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1193, in patched
    arg = patching.__enter__()
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1268, in __enter__
    original, local = self.get_original()
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1242, in get_original
    "%s does not have the attribute %r" % (target, name)
AttributeError: <module 'library.degoss' from '/Users/nicholas/src/ansible-role-degoss/library/degoss.py'> does not have the attribute 'open'

======================================================================
ERROR: test_run_tests_failure (__main__.DegossTestCase)
Tests that degoss can handle failed tests appropriately.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1193, in patched
    arg = patching.__enter__()
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1268, in __enter__
    original, local = self.get_original()
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1242, in get_original
    "%s does not have the attribute %r" % (target, name)
AttributeError: <module 'library.degoss' from '/Users/nicholas/src/ansible-role-degoss/library/degoss.py'> does not have the attribute 'open'

======================================================================
ERROR: test_run_tests_success (__main__.DegossTestCase)
Tests that degoss can handle successful tests appropriately.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1193, in patched
    arg = patching.__enter__()
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1268, in __enter__
    original, local = self.get_original()
  File "/Users/nicholas/.virtualenvs/otcas27/lib/python2.7/site-packages/mock.py", line 1242, in get_original
    "%s does not have the attribute %r" % (target, name)
AttributeError: <module 'library.degoss' from '/Users/nicholas/src/ansible-role-degoss/library/degoss.py'> does not have the attribute 'open'

----------------------------------------------------------------------
Ran 23 tests in 0.036s

FAILED (errors=4)
```

### PR DESCRIPTION
**Pull request by nicholasamorim**
Tests now use mock_open and pass on Python 3.7 as well.

Fixed #45

### CODE REVIEW

File: tests.py [hunk 1]
```diff
@@ -445,15 +439,22 @@ def test_run_tests_failure(self, mock_new_popen, mock_fail, mock_fopen):
...
         mock_new_popen.return_value = mock_process
 
-        mock_file = mock.MagicMock()
-        mock_file_enterer = mock.MagicMock()
-        mock_file_enterer.__enter__.return_value = mock_file
-        mock_fopen.return_value = mock_file_enterer
+        # mock_file = mock.MagicMock()
+        # mock_file_enterer = mock.MagicMock()
+        # mock_file_enterer.__enter__.return_value = mock_file
+        # mock_fopen.return_value = mock_file_enterer
```
Comment (naftulikay) [inline tests.py]: Can you drop these lines? Otherwise LGTM.

**Review by naftulikay — CHANGES_REQUESTED** [review]

**Review by naftulikay — APPROVED** [review]
LGTM :+1:

Please squash and then I'll merge.

File: tests.py (hunk 1 above)
Comment (nicholasamorim) [reply tests.py]: Done.

**Comment by nicholasamorim** [comment]
Done, squashed them

**Comment by naftulikay** [comment]
Released as [v3.0.3](https://github.com/naftulikay/ansible-role-degoss/releases/tag/v3.0.3).

**Pull request merged**